src/hooks/useAppDataQuery.js

```diff
@@ -0,0 +1,21 @@
+import { useDataQuery } from '@dhis2/app-service-data'
+
+const query = {
+    me: {
+        resource: 'me',
+    },
+}
+
+const useAppDataQuery = () => {
+    const { loading, error, data } = useDataQuery(query)
+
+    return {
+        loading,
+        error,
+        data: {
+            userAuthorities: data?.me?.authorities ?? [],
+        },
```

Comment:
I think we should stick to the pattern being used by `useDataQuery` here. If the local `data` const is `undefined`, the `data` property we return should also be `undefined`. The current way also makes sense, but it's just a little inconsistent with how the normal dataQuery works.

I've also done a little bit of research into how the `dataQuery` works internally. The `data` object will only be populated when **all** requests have been resolved successfully. Whenever one request fails, `data` will be undefined and `error` will be populated. As such, we can safely assume that once `data` is an object, the `data.me` property is also available.

So I would suggest changing the highlighted code to:
```javascript
        data: data ? { userAuthorities: data.me.authorities } : undefined,
```